src/app/components/media/CreateMediaSource.js

```diff
@@ -0,0 +1,493 @@
+    newLinks.splice(index, 1);
+    setLinks(newLinks);
+  };
+
+  const handleAddLink = () => {
+    const newLinks = links.slice();
+    newLinks.push({ url: '', error: '' });
+    setLinks(newLinks);
+  };
+
+  const validatePrimaryLink = () => {
+    const linkify = new LinkifyIt();
+    if (primaryUrl.url.trim()) {
+      const validateUrl = linkify.match(primaryUrl.url);
+      if (Array.isArray(validateUrl) && validateUrl[0] && validateUrl[0].url) {
+        return true;
+      }
+      const error = (
+        <FormattedMessage
+          id="sourceInfo.invalidLink"
+          defaultMessage="Please enter a valid URL"
+          description="Error message for invalid link"
+        />
+      );
+      setPrimaryUrl({ url: primaryUrl.url, error });
+      return false;
+    }
+    return true;
+  };
+
+  const validateLinks = () => {
+    const linkify = new LinkifyIt();
+
+    let success = true;
+
+    const newLinks = links.slice().filter(link => !!link.url.trim());
+
+    newLinks.forEach((item_) => {
+      const item = item_;
+      const url = linkify.match(item.url);
+      if (Array.isArray(url) && url[0] && url[0].url) {
+        item.url = url[0].url;
+      } else {
+        item.error = (
+          <FormattedMessage
+            id="sourceInfo.invalidLink"
+            defaultMessage="Please enter a valid URL"
+            description="Error message for invalid link"
+          />
+        );
+        success = false;
+      }
+    });
+
+    setLinks(newLinks);
+    return success;
+  };
+
+  const handleCancelOrSave = () => {
+    onCancel();
+  };
+
+  const handleCancelDialog = () => {
+    setDialogOpen(false);
+    setValidatePrimaryLinkExist(false);
+  };
+
+  const handleSubmitDialog = () => {
+    relateToExistingSource({ dbid: existingSource.id });
+  };
+
+  const handleSave = () => {
+    if (!submitDisabled && validateLinks() && validatePrimaryLink()) {
+      setSubmitDisabled(true);
+      const urls = [];
+      let newLinks = [primaryUrl].concat(links);
+      newLinks = newLinks.filter(link => !!link.url.trim());
+      newLinks.forEach((link) => {
+        urls.push(link.url);
+      });
+
+      const onFailure = (transaction) => {
+        setSubmitDisabled(false);
+        const error = getErrorObjects(transaction);
+        if (Array.isArray(error) && error.length > 0) {
+          if (error[0].code === CheckError.codes.DUPLICATED) {
+            setDialogOpen(true);
+            setExistingSource(error[0].data);
+          } else {
+            const messageError = getErrorMessage(transaction, <GenericUnknownErrorMessage />);
+            setMessage(messageError);
+          }
+        }
+      };
+
+      const onSuccess = () => {
+        handleCancelOrSave();
+      };
+
+      Relay.Store.commitUpdate(
+        new CreateSourceMutation({
+          name: sourceName,
+          slogan: sourceName,
+          image,
+          urls,
+          validate_primary_link_exist,
```

Comment:
Please change to `validate_primary_link_exist: validatePrimaryLinkExist,`

Reply:
Done